api/metadata_handler_test.go

```diff
@@ -109,55 +109,36 @@ func TestHandleMetadataTriage_InvalidBody(t *testing.T) {
 	assert.Equal(t, http.StatusBadRequest, w.Code)
 }
 
-func TestFilterMetadataHanlder_Success(t *testing.T) {
-	handler := func(w http.ResponseWriter, r *http.Request) {
-		http.ServeFile(w, r, "../shared/metadata_testdata/gzip_testfile.tar.gz")
-	}
-	server := httptest.NewServer(http.HandlerFunc(handler))
-	defer server.Close()
+func TestMetadataHanlder_GET_Success(t *testing.T) {
```

Comment:
But now we no longer have tests for the fetcher itself including the decompression and deserialization logic. Could you move this test with a server to `metadata_util_test.go`?

Reply:
Good point!